### DISCUSSION TITLE
Custom config key support for base-package

### DISCUSSION THREAD

**Discussion by Nexey** [quarkiverse/quarkus-openapi-generator#711] [General]
It seems that [`quarkus.openapi-generator.codegen.spec.<filename>.base-package`](https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html#getting-started) doesn't support custom config keys yet.
For instance:
![image](https://github.com/quarkiverse/quarkus-openapi-generator/assets/1363147/173c8134-7f5a-4243-857b-5cff412dfbd4)

In this scenario, 

- `quarkus.openapi-generator.codegen.spec.DEFAULT_CONFIG_KEY.base-package=my.custom.package` will properly generate the Api interfaces and the models in the "my.custom.package" package
- `quarkus.openapi-generator.codegen.spec.CUSTOM_CONFIG_KEY.base-package=my.custom.package` will generate the files in the default package (which would be something like "org.openapi.quarkus.DEFAULT_CONFIG_KEY")

Is this expected behavior?

Thanks in advance.

**Comment by ricardozanini** [comment]
Hi! I'm not familiar with custom config keys. What's not working specifically?

**Reply by Nexey** [reply]
Hi, sorry it took me so long to get back to you. I ended up using the generated configKey openapi gives me and everything works fine, but I'll give this another attempt at explaining the issue.

For the working part:
By default, Open API will set the `configKey` to a sanitized version of the file name. If my file is called "`my-openapi-doc-v1.json`", the generated `configKey` will probably end up being set to "`my_openapi_doc_v1_json`".

We can already override that `configKey` using `quarkus.openapi-generator.codegen.my_openapi_doc_v1_json.config-key=foo`. 

The generated interfaces will now properly display `@RegisterRestClient(baseUri="https://github.com/quarkiverse/quarkus-openapi-generator/discussions/711", configKey="foo")`.

For the part that isn't working how I thought it would:
Whenever we want to set a specific property for a generated client, we need to use the `configKey`. If I wanted to return "`Reponse`" Objects for my `my_openapi_doc_v1_json`, I would set `quarkus.openapi-generator.codegen.spec.my_openapi_doc_v1_json .return-response=true`.
What would make sense is supporting the custom `configKey`, which in my example is now "`foo`":
`quarkus.openapi-generator.codegen.spec.foo.return-response=true`.

Using quarkus rest, it would be nice to set the URL using the custom `configKey` as well:
`quarkus.rest-client.foo.url=https://github.com/quarkiverse/quarkus-openapi-generator/discussions/711`.

Sorry if my first post was a bit confusing, I hope this clears some things up. To sum it up: while a `configKey` can be overriden, the new value isn't supported in the property files and the default `configKey` is the only valid one, whereas I was expecting the new `configKey` to be used as a complete replacement for setting other properties.

**Reply by ricardozanini** [reply]
@Nexey has been a long time since I first implemented this feature. But IIRC that should work as you explained. If not, it's a bug. Do you mind sending a PR?
